docs/design/proposals/adoption/resource-adoption.md

```diff
@@ -0,0 +1,225 @@
+
+The proposed k8s adopted tag is `services.k8s.aws/adopted: true`
+
+The adopted annotation would remain on the custom resource for the remainder of its lifecycle, and should not be modified by any other manager operation. 
+
+## AWS Resource Tagging
+
+AWS resources adopted by an ACK controller will have tags retroactively applied (whenever possible) to indicate its association with an ACK deployment. The details of these tags are discussed in another issue ([#148](https://github.com/aws/aws-controllers-k8s/issues/148)). ACK managers should not place any conditions on the existence of ACK tags on the AWS resources to gate the adoption process. The manager should should override any existing ACK tags.
+
+## Resource Type Blocklisting
+
+Adoption, for some resources, is not possible due to any number of the following reasons:
+
+* They are ephemeral resources which don’t support additional lifecycle operations (one-time jobs)
+* The service cannot provide adequate information to produce a full specification
+* Unforeseen limitations within ACK or the service directly
+
+For any of these cases, the resource can be marked as “not adoptable” and the validation webhook will subsequently reject any adopted resources of this type. Therefore, the manager will never attempt to run the adoption process. This flag can be set in the generator configuration file with the `cannot_adopt` boolean option supplied for any resource.
+
+## Adoption Process Diagram
+
+![sequence-diagram](./images/adoption-sequence-diagram.png)
+# Potential Alternatives
+
+## Resource Adoption Annotation
+
+The existing custom resource definitions (every service resource) would accept an additional annotation that the manager interprets as an indication that the resource specification should be ignored and instead that it should be formed through adoption. For any resource marked with this annotation, upon creation, the manager would attempt to describe the AWS resource associated with the annotated value. If none such resource exists, the manager would update the state of the custom resource into a `AdoptionFailed` terminal state. If the resource does exist, the manager would continue the follow the same adoption process as defined in the [Proposed Implementation](https://quip-amazon.com/lXxVAKU9dv16#SZW9CAMeXE1) section.
+
+### Example Specification
+
+**An example adopted S3 resource**
+
+```
+apiVersion: s3.services.k8s.aws/v1alpha1
+kind: Bucket
+metadata:
+  annotations:
+    services.k8s.aws/arn: arn:aws:s3:::my-existing-bucket
+  ...
+spec:
+   ...
+```
+
+### Notable Pitfalls
```

Comment:
In addition to these pitfalls, ACK could adopt a resource that it only has read access to. Reconciliation via describe calls would work, but in the event of a user trying to mutate via ACK, permissions errors would be encountered.  I suppose this could happen to resources even if they are created by ACK if there are only partial write permissions and an update happens later on. Just noting it here, although I don't think it's a huge deal.